Resolving relative refs broken with Spring Boot 3.2

Parsing schemas with refs to other files fails with the following exception, when the jar file is run:

```
08:35:32.381 [main] WARN io.swagger.v3.parser.OpenAPIV3Parser -- Exception while resolving:
java.lang.RuntimeException: Unable to load RELATIVE ref: ./error.openapi.yaml path: /specs
        at io.swagger.v3.parser.util.RefUtils.readExternalRef(RefUtils.java:220)
        at io.swagger.v3.parser.ResolverCache.loadRef(ResolverCache.java:156)
        at io.swagger.v3.parser.processors.ExternalRefProcessor.processRefToExternalSchema(ExternalRefProcessor.java:89)
        at io.swagger.v3.parser.processors.SchemaProcessor.processReferenceSchema(SchemaProcessor.java:238)
        at io.swagger.v3.parser.processors.SchemaProcessor.processSchema(SchemaProcessor.java:59)
        at io.swagger.v3.parser.processors.ResponseProcessor.processResponse(ResponseProcessor.java:56)
        at io.swagger.v3.parser.processors.OperationProcessor.processOperation(OperationProcessor.java:86)
        at io.swagger.v3.parser.processors.PathsProcessor.processPaths(PathsProcessor.java:88)
        at io.swagger.v3.parser.OpenAPIResolver.resolve(OpenAPIResolver.java:72)
        at io.swagger.v3.parser.OpenAPIResolver.resolve(OpenAPIResolver.java:59)
        at io.swagger.v3.parser.OpenAPIV3Parser.resolve(OpenAPIV3Parser.java:240)
        at io.swagger.v3.parser.OpenAPIV3Parser.readContents(OpenAPIV3Parser.java:183)
        at io.swagger.v3.parser.OpenAPIV3Parser.readLocation(OpenAPIV3Parser.java:97)
        at io.swagger.v3.parser.OpenAPIV3Parser.read(OpenAPIV3Parser.java:124)
        at com.example.demo.DemoApplication.main(DemoApplication.java:17)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:91)
        at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:53)
        at org.springframework.boot.loader.launch.JarLauncher.main(JarLauncher.java:58)
Caused by: java.lang.UnsupportedOperationException: null
        at jdk.zipfs/jdk.nio.zipfs.ZipPath.toFile(ZipPath.java:669)
        at io.swagger.v3.parser.util.RefUtils.readAll(RefUtils.java:228)
        at io.swagger.v3.parser.util.RefUtils.readExternalRef(RefUtils.java:191)
        ... 21 common frames omitted
08:35:32.383 [main] INFO io.swagger.v3.parser.OpenAPIV3Parser -- io.swagger.v3.parser.OpenAPIV3Parser@75c56eb9: attribute paths.'/pets'(get).responses.default.content.'application/json'.schema.#/components/schemas/Error is missing
08:35:32.384 [main] INFO io.swagger.v3.parser.OpenAPIV3Parser -- io.swagger.v3.parser.OpenAPIV3Parser@75c56eb9: attribute paths.'/pets'(post).responses.default.content.'application/json'.schema.#/components/schemas/Error is missing
08:35:32.384 [main] INFO io.swagger.v3.parser.OpenAPIV3Parser -- io.swagger.v3.parser.OpenAPIV3Parser@75c56eb9: Unable to load RELATIVE ref: ./error.openapi.yaml path: /specs
```

I'm using Spring Boot 3.2.x which has a nested jar and the loader has been re-written in Spring Boot 3.2: [Spring Boot 3.2 Release Notes: Nested Jar Support](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.2-Release-Notes#nested-jar-support). 
This seems to break the OpenAPI Parsers way of resolving $refs.

--- a/modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/RefUtils.java
+++ b/modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/RefUtils.java
@@ -6,7 +6,6 @@ import io.swagger.v3.parser.processors.ExternalRefProcessor;
 import io.swagger.v3.parser.urlresolver.PermittedUrlsChecker;
 import org.apache.commons.io.IOUtils;
 
-import java.io.FileInputStream;
 import java.io.IOException;
 import java.io.InputStream;
 import java.net.URL;
@@ -235,7 +234,7 @@ public class RefUtils {
     }
 
     private static String readAll(Path path) throws IOException {
-        try (InputStream inputStream = new FileInputStream(path.toFile())) {
+        try (InputStream inputStream = Files.newInputStream(path)) {
             return IOUtils.toString(inputStream, UTF_8);
         }
     }

--- a/modules/swagger-parser-v3/src/test/java/io/swagger/v3/parser/util/RefUtilsTest.java
+++ b/modules/swagger-parser-v3/src/test/java/io/swagger/v3/parser/util/RefUtilsTest.java
@@ -195,10 +195,6 @@ public class RefUtilsTest {
             times = 1;
             result = true;
 
-            pathToUse.toFile();
-            times = 1;
-            result = file;
-
         }};
     }
 

--- a/modules/swagger-parser-v3/src/test/java/io/swagger/v3/parser/util/RefUtilsZipFileSystemTest.java
+++ b/modules/swagger-parser-v3/src/test/java/io/swagger/v3/parser/util/RefUtilsZipFileSystemTest.java
@@ -1,0 +1,50 @@
+package io.swagger.v3.parser.util;
+
+import io.swagger.v3.parser.models.RefFormat;
+import io.swagger.v3.parser.urlresolver.PermittedUrlsChecker;
+import org.junit.Test;
+
+import java.net.URI;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.Map;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotSame;
+
+public class RefUtilsZipFileSystemTest {
+
+    @Test
+    public void testReadExternalRefFromZipFileSystem() throws Exception {
+        Path archive = Files.createTempFile("ref-utils", ".zip");
+        Files.delete(archive);
+
+        Map<String, String> environment = Collections.singletonMap("create", "true");
+        URI archiveUri = URI.create("jar:" + archive.toUri());
+        String expectedContent = "openapi: 3.0.0\n";
+
+        try (FileSystem zipFileSystem = FileSystems.newFileSystem(archiveUri, environment)) {
+            Path parentDirectory = zipFileSystem.getPath("/specs");
+            Path referencedFile = parentDirectory.resolve("error.openapi.yaml");
+            Files.createDirectories(parentDirectory);
+            Files.write(referencedFile, expectedContent.getBytes(UTF_8));
+
+            assertNotSame(FileSystems.getDefault(), referencedFile.getFileSystem());
+
+            String actualContent = RefUtils.readExternalRef(
+                    "./error.openapi.yaml",
+                    RefFormat.RELATIVE,
+                    Collections.emptyList(),
+                    parentDirectory,
+                    new PermittedUrlsChecker());
+
+            assertEquals(expectedContent, actualContent);
+        } finally {
+            Files.deleteIfExists(archive);
+        }
+    }
+}

